### PR TITLE
Syncapp

### PR DESCRIPTION
**Pull request by abhi-intelops**

### CODE REVIEW

File: pkg/cmd/setup_apps.go [hunk 1]
```diff
@@ -77,9 +77,16 @@ var appsCmd = &cobra.Command{
...
-		//prepare agent proto to push app config
-		//agent store data on cassandra
+		client, err := agent.GetAgentClient(captenConfig)
+		if err != nil {
+			clog.Logger.Errorf("failed to fetch client, err:%v", err)
+			return
+		}
+
+		if err := agent.SaveAppConfigsOnAgent(client, captenConfig.AppsTempDirPath); err != nil {
+			clog.Logger.Errorf("Err while SaveAppConfigsOnAgent: %v", err)
```
Comment (vramk23) [inline pkg/cmd/setup_apps.go:87]: pass captenConfig object to method, it need to build the complete path

**Review by vramk23 — CHANGES_REQUESTED** [review]

File: pkg/cmd/setup_apps.go [hunk 2]
```diff
@@ -77,9 +77,16 @@ var appsCmd = &cobra.Command{
...
 		}
 
-		//push the app config to cluster
-		//prepare agent proto to push app config
-		//agent store data on cassandra
+		client, err := agent.GetAgentClient(captenConfig)
+		if err != nil {
+			clog.Logger.Errorf("failed to fetch client, err:%v", err)
+			return
+		}
```
[the remark this answers is not in the document]
Comment (abhi-intelops) [reply pkg/cmd/setup_apps.go:84]: dibe

File: pkg/cmd/setup_apps.go (hunk 1 above)
Comment (abhi-intelops) [reply pkg/cmd/setup_apps.go:81]: done

File: pkg/types/types.go [hunk 3]
```diff
@@ -50,3 +56,39 @@ type ClusterInfo struct {
...
+		return agentpb.SyncAppData{}, err
+	}
+
+	return agentpb.SyncAppData{
+		Config: &agentpb.AppConfig{
+			ReleaseName:         a.ReleaseName,
+			AppName:             a.Name,
+			Version:             a.Version,
+			Category:            "",
+			Description:         "",
```
[the remark this answers is not in the document]
Comment (abhi-intelops) [reply pkg/types/types.go]: done

File: pkg/types/types.go [hunk 4]
```diff
@@ -50,3 +56,39 @@ type ClusterInfo struct {
...
+			ChartName:           a.ChartName,
+			RepoName:            a.RepoName,
+			RepoURL:             a.RepoURL,
+			Namespace:           a.Namespace,
+			CreateNamespace:     a.CreateNamespace,
+			PrivilegedNamespace: a.PrivilegedNamespace,
+			Icon:                []byte{},
+			LaunchURL:           "",
```
[the remark this answers is not in the document]
Comment (abhi-intelops) [reply pkg/types/types.go]: done

**Pull request merged**